src/System.Net.Security/src/System/Net/Security/SslServerAuthenticationOptions.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Security
+{
+    public class SslServerAuthenticationOptions
+    {
+        private X509Certificate _serverCertificate;
+        private X509RevocationMode _checkCertificateRevocation = X509RevocationMode.NoCheck;
+        private SslProtocols _enabledSslProtocols = SecurityProtocol.SystemDefaultSecurityProtocols;
+        private EncryptionPolicy _encryptionPolicy = EncryptionPolicy.RequireEncryption;
+
+        internal RemoteCertValidationCallback _certValidationDelegate;
+
+        public bool AllowRenegotiation { get; set; }
+
+        public bool ClientCertificateRequired { get; set; }
+
+        public List<SslApplicationProtocol> ApplicationProtocols { get; set; }
+
+        public RemoteCertificateValidationCallback RemoteCertificateValidationCallback { get; set; }
+
+        public X509Certificate ServerCertificate
+        {
+            get => _serverCertificate;
+            set => _serverCertificate = value;
+        }
+
+        public SslProtocols EnabledSslProtocols
+        {
+            get => _enabledSslProtocols;
+            set => _enabledSslProtocols = value;
+        }
```

Comment:
ServerCertificate and EnabledSslProtocols could both be auto-properties now, if you wanted.

Reply:
enabledsslprotocols has a default, left that as if, changed the rest to auto properties.